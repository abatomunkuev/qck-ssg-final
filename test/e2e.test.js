/* eslint-disable no-undef */
import { run } from "./run";
import fs from "fs";

describe("end-to-end integration", () => {
  test("prints error and help message when no arguments given", async () => {
    const { stderr, stdout, exitCode } = await run();
    expect(exitCode).toBe(1);
    expect(stderr).toMatchSnapshot();
    expect(stdout).toEqual("[]");
  });

  test("prints error and help message when no arguments given", async () => {
    const { stderr, stdout, exitCode } = await run();
    expect(exitCode).toBe(1);
    expect(stderr).toMatchSnapshot();
    expect(stdout).toEqual("[]");
  });

  test("prints help message when --help given", async () => {
    const { stderr, stdout, exitCode } = await run("--help");
    expect(exitCode).toBe(0);
    expect(stdout).toMatchSnapshot();
    expect(stderr).toEqual("");
  });

  // test("check if file created in the dist directory ", () => {
  //   expect(
  //     fs
  //       .exists("/dist/index.html", (exists) => {
  //         return exist;
  //       })
  //       .not.toBeTruthy()
  //   );
  // });
  // test("prints formatted output to specified width for --indent", async () => {
  //   const { stderr, stdout, exitCode } = await run(
  //     "--indent",
  //     "4",
  //     "./test/sample.html"
  //   );
  //   expect(exitCode).toBe(0);
  //   expect(stdout).toMatchSnapshot();
  //   expect(stderr).toEqual("");
  // });

  // test("prints formatted output to specified width for -i", async () => {
  //   const { stderr, stdout, exitCode } = await run(
  //     "-i",
  //     "4",
  //     "./test/sample.html"
  //   );
  //   expect(exitCode).toBe(0);
  //   expect(stdout).toMatchSnapshot();
  //   expect(stderr).toEqual("");
  // });
});
