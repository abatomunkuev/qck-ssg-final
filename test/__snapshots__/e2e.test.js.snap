// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`end-to-end integration prints error and help message when no arguments given 1`] = `
"file:///Users/anatoliyserputov/QckStaticSiteGenerator/main.js:94
    if (file.input.includes(\\".\\")) {
                   ^

TypeError: Cannot read property 'includes' of undefined
    at main (file:///Users/anatoliyserputov/QckStaticSiteGenerator/main.js:94:20)
    at Main (file:///Users/anatoliyserputov/QckStaticSiteGenerator/main.js:117:3)
    at callMain (file:///Users/anatoliyserputov/QckStaticSiteGenerator/options.js:27:5)
    at Options (file:///Users/anatoliyserputov/QckStaticSiteGenerator/options.js:30:3)
    at file:///Users/anatoliyserputov/QckStaticSiteGenerator/index.js:7:1
    at ModuleJob.run (internal/modules/esm/module_job.js:183:25)
    at async Loader.import (internal/modules/esm/loader.js:178:24)
    at async Object.loadESM (internal/process/esm_loader.js:68:5)"
`;

exports[`end-to-end integration prints error and help message when no arguments given 2`] = `
"file:///Users/anatoliyserputov/QckStaticSiteGenerator/main.js:94
    if (file.input.includes(\\".\\")) {
                   ^

TypeError: Cannot read property 'includes' of undefined
    at main (file:///Users/anatoliyserputov/QckStaticSiteGenerator/main.js:94:20)
    at Main (file:///Users/anatoliyserputov/QckStaticSiteGenerator/main.js:117:3)
    at callMain (file:///Users/anatoliyserputov/QckStaticSiteGenerator/options.js:27:5)
    at Options (file:///Users/anatoliyserputov/QckStaticSiteGenerator/options.js:30:3)
    at file:///Users/anatoliyserputov/QckStaticSiteGenerator/index.js:7:1
    at ModuleJob.run (internal/modules/esm/module_job.js:183:25)
    at async Loader.import (internal/modules/esm/loader.js:178:24)
    at async Object.loadESM (internal/process/esm_loader.js:68:5)"
`;

exports[`end-to-end integration prints formatted output to specified width for --indent 1`] = `""`;

exports[`end-to-end integration prints formatted output to specified width for -i 1`] = `""`;

exports[`end-to-end integration prints help message when --help given 1`] = `
"Usage: -h <file>

Options:
      --help    Show help                                              [boolean]
  -i, --input   File name                                               [string]
  -c, --config  Configuration file                                      [string]
  -v            version                                                [boolean]"
`;

exports[`end-to-end integration when argument --help is specified should print help message 1`] = `
"Usage: -h <file>

Options:
      --help    Show help                                              [boolean]
  -i, --input   File name                                               [string]
  -c, --config  Configuration file                                      [string]
  -v            version                                                [boolean]"
`;

exports[`prints error and help message when no arguments given 1`] = `
"file:///Users/anatoliyserputov/QckStaticSiteGenerator/main.js:94
    if (file.input.includes(\\".\\")) {
                   ^

TypeError: Cannot read property 'includes' of undefined
    at main (file:///Users/anatoliyserputov/QckStaticSiteGenerator/main.js:94:20)
    at Main (file:///Users/anatoliyserputov/QckStaticSiteGenerator/main.js:117:3)
    at callMain (file:///Users/anatoliyserputov/QckStaticSiteGenerator/options.js:27:5)
    at Options (file:///Users/anatoliyserputov/QckStaticSiteGenerator/options.js:30:3)
    at file:///Users/anatoliyserputov/QckStaticSiteGenerator/index.js:7:1
    at ModuleJob.run (internal/modules/esm/module_job.js:183:25)
    at async Loader.import (internal/modules/esm/loader.js:178:24)
    at async Object.loadESM (internal/process/esm_loader.js:68:5)"
`;

exports[`prints help message when --help given 1`] = `
"Usage: -h <file>

Options:
      --help    Show help                                              [boolean]
  -i, --input   File name                                               [string]
  -c, --config  Configuration file                                      [string]
  -v            version                                                [boolean]"
`;
